packages/components/src/modal/test/index.tsx

```diff
@@ -129,4 +129,39 @@ describe( 'Modal', () => {
 			screen.getByText( 'A sweet button', { selector: 'button' } )
 		).toBeInTheDocument();
 	} );
+
+	describe( 'Focus handling', () => {
+		it( 'should focus the first focusable element in the contents when `firstElement` passed as value for `focusOnMount` prop', async () => {
+			const user = userEvent.setup();
+			const FocusMountDemo = () => {
+				const [ isShown, setIsShown ] = useState( false );
+				return (
+					<>
+						<button onClick={ () => setIsShown( true ) }>📣</button>
+						{ isShown && (
+							<Modal
+								focusOnMount="firstElement"
+								onRequestClose={ () => setIsShown( false ) }
+							>
+								<p>Modal content</p>
+								<a
+									href="https://wordpress.org"
+									data-testid="button-with-focus"
+								>
+									Button
+								</a>
+							</Modal>
+						) }
+					</>
+				);
+			};
+			render( <FocusMountDemo /> );
+
+			const opener = screen.getByRole( 'button' );
+
+			await user.click( opener );
+
+			expect( screen.getByTestId( 'button-with-focus' ) ).toHaveFocus();
```

Comment:
> I think that we should add a couple of unit tests to certify the behaviour:

@ciampo This test currently fails because this conditional for `isVisible` fails:

https://github.com/WordPress/gutenberg/blob/b05bee7e2e8eefb2be06c228002e4f9783847747/packages/dom/src/focusable.js#L107-L109

I believe this is because it relies on layout properties of `Element` which are not set in the test envionrment

https://github.com/WordPress/gutenberg/blob/b05bee7e2e8eefb2be06c228002e4f9783847747/packages/dom/src/focusable.js#L55-L61

Is there any prior art for this?

Reply:
Would [this example](https://github.com/WordPress/gutenberg/blob/bb8cf8a18983af8b89b251a07a58f1eae97e3a46/packages/components/src/navigable-container/test/navigable-menu.tsx#L31-L46) help?

Reply:
Ok I've managed to mock `HTMLElement.prototype` to define the values required by the `isVisible` check. I've included a clear comment to explain what I'm doing.

Reply:
Cool! We just wrote a comment at the same time :D In the past I managed to get around this by only mocking `getClientRects` (see my comment above)

Reply:
I've moved it all to `describe` in `beforeEach` and `afterEach`.

Reply:
> I managed to get around this by only mocking getClientRects (see my comment above)

Yes that's a lot simpler. I realised that it's an `OR` condition so we only need a single mock now. I've normalized towards your example.